### ISSUE TITLE
Unable to add an item to a list using overlays in IstioOperator manifest

### ISSUE THREAD

**Issue by ashevtsov-wawa** [istio/istio#34024]
**Bug description**
I am trying to add `--grpcAddr=""` start-up option as per https://istio.io/latest/docs/ops/best-practices/security/#control-plane
To do that I tried to use overlays option in Istio Operator as per https://github.com/istio/istio/blob/da6178604559bdf2c707a57f452d16bee0de90c8/operator/pkg/patch/patch.go#L68
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
...
spec:
  components:
    pilot:
      enabled: true
      k8s:
        overlays:
        - kind: Deployment
          name: istiod
          patches:
          - path: spec.template.spec.containers.[name:discovery].args
            value: --grpcAddr=""
```

The istiod deployment doesn't get patched and here is what I see in the operator's log
```
2021-07-14T01:07:57.673719Z     info    installer       Updating Deployment/istio-system/istiod (istiocontrolplane/)
2021-07-14T01:07:57.680219Z     error   installer       Deployment in version "v1" cannot be handled as a Deployment: v1.Deployment.Spec: v1.DeploymentSpec.Template: v1.PodTemplateSpec.Spec: v1.PodSpec.Containers: []v1.Container: v1.Container.Args: []string: decode slice: expect [ or n, but found ", error found in #10 byte of ...|[{"args":"--grpcAddr|..., bigger context ...|mponent":"Pilot"}},"spec":{"containers":[{"args":"--grpcAddr=\"\"","env":[{"name":"REVISION","value"|...
```

[] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**
`--grpcAddr=""` switch gets added to `args` list in the istiod deployment.

**Steps to reproduce the bug**
Use the istio operator manifest above

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)
```
❯ istioctl version --remote
client version: 1.8.6
control plane version: 1.8.6
data plane version: 1.8.6 (12 proxies)
❯ kubectl version --short
Client Version: v1.21.2
Server Version: v1.19.8-eks-96780e
```

**How was Istio installed?**
Istio Operator

**Environment where the bug was observed (cloud vendor, OS, etc)**
AWS EKS

**Comment by litong01** [comment]
@ashevtsov-wawa just confirmed this is indeed an issue (sort of). This issue is more of a document problem. You probably can do something like this to avoid the issue.

```
          patches:
          - path: spec.template.spec.containers.[name:discovery].args.[30]
            value: --grpcAddr=""
```
The trick is to add a bracket `[]` to let the code know we are dealing with an list, not a hash. Use a big number like 30 is to indicate this is not to replace any existing item. Obviously the number 30 is a bit misleading, but this is the only way to go around this problem.

**Comment by litong01** [comment]
@ashevtsov-wawa Here is a complete yaml file that you can use to fix the issue.

```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: istiod
  namespace: istio-system
spec:
  components:
    pilot:
      enabled: true
      k8s:
        overlays:
        - kind: Deployment
          name: istiod
          patches:
          - path: spec.template.spec.containers.[name:discovery].args.[100]
            value: --grpcAddr=""
```

**Comment by litong01** [comment]
Even more complete example to avoid this kind of issues.
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  name: istiod
  namespace: istio-system
spec:
  components:
    pilot:
      enabled: true
      k8s:
        overlays:
        - kind: Deployment
          name: istiod
          patches:
          - path: spec.template.spec.containers.[name:discovery].args.[100]
            value: --grpcAddr=:15020
          - path: spec.template.spec.containers.[name:discovery].ports.[containerPort:15010]
            value:
              containerPort: 15020
              protocol: TCP
        - kind: Service
          name: istiod
          patches:
          - path: spec.ports.[port:15010]
            value:
              port: 15020
              name: grpc-xds
              protocol: TCP
```

**Comment by ashevtsov-wawa** [comment]
Thank you!
Should I keep this ticket open until documentation gets updated?

**Comment by litong01** [comment]
@ashevtsov-wawa I will submit a PR to update the comments in the source code file. please review after the PR is submitted. Thanks.

**Comment by litong01** [comment]
Fixed by #35044
